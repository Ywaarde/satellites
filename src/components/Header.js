import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <div>
      <h1>Satellites</h1>
      <NavLink to='/' activeClassName='is-active' exact={true}>Misson Overview</NavLink>
      <NavLink to='/create' activeClassName='is-active'>Add mission</NavLink>
    </div>
)

export default Header;