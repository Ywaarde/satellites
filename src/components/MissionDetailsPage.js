import React from 'react';

const MissionDetailsPage = (props) => (
    <div>
    This is the page with details for Mission ID: {props.match.params.id}.
    </div>
);

export default MissionDetailsPage;