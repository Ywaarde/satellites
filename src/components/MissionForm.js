import React from 'react';

export default class MissionForm extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      MissionID: props.mission ? props.mission.mission_id : props.new_id,
      MissionNameShort: props.mission ? props.mission.name_short : '',
      MissionNameFull: props.mission ? props.mission.name_full : '',
      MissionAgencies: props.mission ? props.mission.agencies : '',
      MissionStatus: props.mission ? props.mission.status : ''
    };
  }

  onShortNameChange = (e) => {
    const MissionNameShort = e.target.value;
    this.setState({ MissionNameShort });
  }

  onFullNameChange = (e) => {
    const MissionNameFull = e.target.value;
    this.setState({ MissionNameFull });
  }

  onAgenciesChange = (e) => {
    const MissionAgencies = e.target.value;
    this.setState({ MissionAgencies });
  }

  onMissionStatusChange = (e) => {
    const MissionStatus = e.target.value;
    this.setState({ MissionStatus });
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.props.onSubmit(this.state);
  }

  render() {
    return (
      <div>
        {this.props.mission && <h3>Mission ID {this.props.mission.mission_id}</h3>}
        {this.props.new_id && <h3>Mission ID {this.props.new_id}</h3>}
        <form onSubmit={this.onSubmit}>
          <input
            type='text'
            placeholder='Mission Short Name'
            required={true}
            autoFocus
            value={this.state.MissionNameShort}
            onChange={this.onShortNameChange}
          />
          <textarea
            placeholder='Mission Full Name'
            required={true}
            value={this.state.MissionNameFull}
            onChange={this.onFullNameChange}
          >
          </textarea>
          <input
            type='text'
            placeholder='Mission Agencies'
            required={true}
            value={this.state.MissionAgencies}
            onChange={this.onAgenciesChange}
          />
          <input
            type='text'
            placeholder='Mission Status'
            required={true}
            value={this.state.MissionStatus}
            onChange={this.onMissionStatusChange}
          />
          <button>Save Mission</button>
        </form>
      </div>
    );
  }
}

// Full mission details
    // Mission ID*
    // Mission Name Short*
    // Mission Name Full*
    // Mission Agencies*
    // Mission MissionStatus*
    // Launch Date,
    // EOL Date,
    // Applications,
    // Instruments,
    // Instrument IDs,
    // Orbit Type,
    // Orbit Altitude,
    // Orbit Period,
    // Orbit Inclination,
    // Repeat Cycle,
    // Orbit LST,
    // Orbit Longitude,
    // Orbit Sense,
    // Mission Site,
    // Data Access Portal
