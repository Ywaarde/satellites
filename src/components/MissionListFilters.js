import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByMissionID, sortByMissionShortName, sortByMissionAgencies, sortByMissionStatus } from '../actions/filters';

export class MissionListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = (e) => {
    switch (e.target.value) {
      case 'mission_id':
        this.props.sortByMissionID();
        break;
      case 'name_short':
        this.props.sortByMissionShortName();
        break;
      case 'agencies':
        this.props.sortByMissionAgencies();
        break;
      case 'status':
        this.props.sortByMissionStatus();
        break;
      default:
        break;
    }
  }

  render() {
    return (
      <div>
        <p>Search for 'Mission Name Short'</p>
        <input
          type='text'
          option={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <p>Sort by: </p><select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value='mission_id'>Mission ID</option>
          <option value='name_short'>Mission Name Short</option>
          <option value='agencies'>Mission Agencies</option>
          <option value='status'>Mission Status</option>
        </select>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByMissionID: () => dispatch(sortByMissionID()),
  sortByMissionShortName: () => dispatch(sortByMissionShortName()),
  sortByMissionAgencies: () => dispatch(sortByMissionAgencies()),
  sortByMissionStatus: () => dispatch(sortByMissionStatus())
});

export default connect(mapStateToProps, mapDispatchToProps)(MissionListFilters);
