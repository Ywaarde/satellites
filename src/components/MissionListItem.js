import React from 'react';
import { Link } from 'react-router-dom';

const MissionListItem = ({ id, mission_id, name_short, agencies, status }) => (
    <tr>
      <td>{mission_id}</td>
      <td>{name_short}</td>
      <td>{agencies}</td>
      <td>{status}</td>
      <td>
        <Link to={`/edit/${mission_id}`}>Edit Mission</Link>
      </td>
    </tr>
);

export default MissionListItem;