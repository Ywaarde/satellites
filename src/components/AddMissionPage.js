import React from 'react';
import { connect } from 'react-redux';
import MissionForm from './MissionForm';
import { addMission } from '../actions/missions';

export class AddMissionPage extends React.Component {
  onSubmit = (mission) => {
    this.props.addMission(mission);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <h3>Add New Mission</h3>
        <MissionForm
          new_id = {this.props.new_id}
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  const highest_id = state.missions.reduce((prev, current) => (prev.mission_id > current.mission_id) ? prev : current);
  return {
      new_id: highest_id.mission_id + 1
  };
};

const mapDispatchToProps = (dispatch) => ({
  addMission: (mission) => dispatch(addMission(mission))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddMissionPage);