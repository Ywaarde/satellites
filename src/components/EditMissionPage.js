import React from 'react';
import { connect } from 'react-redux';
import MissionForm from './MissionForm';
import { editMission, removeMission } from '../actions/missions';

export class EditMissionPage extends React.Component {
  onSubmit = (mission) => {
    this.props.editMission(this.props.mission.id, mission);
    this.props.history.push('/');
  }
  onClick = () => {
    this.props.removeMission(this.props.mission.id);
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <MissionForm
          mission = {this.props.mission}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onClick}
        >Delete Mission</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  mission: state.missions.find((mission) => mission.mission_id.toString() === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editMission: (id, mission) => dispatch(editMission(id, mission)),
  removeMission: (id) => dispatch(removeMission({ id }))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditMissionPage);