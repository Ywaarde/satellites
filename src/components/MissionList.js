import React from 'react';
import { connect } from 'react-redux';
import MissionListItem from './MissionListItem';
import getVisibleMissions from '../selectors/missions';

export const MissionList = (props) => (
  <div>
    <h3>Mission List</h3>
    {props.missions.length === 0 && <p>No missions to display</p>}
    <table>
      <thead>
        <tr>
          <th>Mission ID</th>
          <th>Mission Name Short</th>
          <th>Mission Agencies</th>
          <th>Mission Status</th>
        </tr>
      </thead>
      <tbody>
        {props.missions.map((mission) => (
          <MissionListItem key={mission.id} {...mission}
          />
        ))}
      </tbody>
    </table>
  </div>
);

const mapStateToProps = (state) => {
  return {
    missions: getVisibleMissions(state.missions, state.filters)
  };
};

export default connect(mapStateToProps)(MissionList);

