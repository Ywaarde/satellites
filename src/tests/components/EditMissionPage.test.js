import React from 'react';
import { shallow } from 'enzyme';
import testMissions from '../fixtures/missions';
import { EditMissionPage } from '../../components/EditMissionPage';

let editMission, removeMission, history, wrapper

beforeEach(() => {
    editMission = jest.fn()
    removeMission = jest.fn()
    history = { push: jest.fn() }
    wrapper = shallow(
        <EditMissionPage 
            editMission={editMission} 
            removeMission={removeMission} 
            history={history} 
            mission={testMissions[0]} 
        />
    )
})

test('should render EditMissionPage correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('should handle editMission', () => {
    const editedMission = {
        ...testMissions[0],
        status: 'Different'
    }
    wrapper.find('MissionForm').prop('onSubmit')(wrapper.find('MissionForm').prop('mission'))
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(editMission).toHaveBeenLastCalledWith(testMissions[0].id, testMissions[0])
})

test('should handle removeMission', () => {
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(removeMission).toHaveBeenLastCalledWith(testMissions[0].id)
})