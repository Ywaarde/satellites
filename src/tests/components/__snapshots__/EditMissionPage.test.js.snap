// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditMissionPage correctly 1`] = `
<div>
  <MissionForm
    mission={
      Object {
        "agencies": "Fspace",
        "id": "123abc",
        "mission_id": 1,
        "name_full": "Mission to Mars",
        "name_short": "Mars",
        "status": "Approved",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Delete Mission
  </button>
</div>
`;
